src/atomicskiplist.ml

```diff
@@ -0,0 +1,242 @@
+type 'a markablereference = { node : 'a; marked : bool }
+(** markable reference: stores a reference to a node and has a field to specify if it is marked *)
+
+type 'a node = {
+  key : int;
+  height : int;
+  next : 'a node option markablereference Atomic.t array;
+}
+
+type 'a t = { head : 'a node }
+
+let max_height = 10
+
+(** create_new_node: creates a new node with some value and height *)
+let create_new_node value height =
+  let next =
+    let arr =
+      Array.make (max_height + 1) (Atomic.make { node = None; marked = false })
+    in
+    for level = 1 to max_height do
+      arr.(level) <- Atomic.make { node = None; marked = false }
+    done;
```

Comment:
The reason I used the for loop to initialize the array is so that all the elements of the array would be physically different. A change to one of the elements of the array should not reflect in the other elements. I suppose there might be a more idiomatic way of achieving this.

Reply:
`Array.init` ?

Reply:
I see. Yes, `Array.init` should do the trick.